### ISSUE TITLE
`plot_error_heatmap` lead-time axis label shows a wrong unit for sub-hourly or indivisible step lengths

### ISSUE THREAD

**Issue by nikhil3495** [mllam/neural-lam#742]
## Description

`plot_error_heatmap` (`neural_lam/vis.py`) builds the x-axis label from only the first character of the unit name:

```python
time_step_int, time_step_unit = utils.get_integer_time(step_length)
...
ax.set_xlabel(f"Lead time ({time_step_unit[0]})", size=layout["tick_label_size"])
```

`utils.get_integer_time` returns the unit as a full word - `weeks`, `days`, `hours`, `minutes`, `seconds`, `milliseconds`, `microseconds`, or `unknown`. `[0]` only happens to be right for `weeks`/`days`/`hours`/`seconds`. It is wrong for:

| `step_length` | `time_step_unit` | label shown | should be |
|---|---|---|---|
| `timedelta(minutes=15)` | `minutes` | `Lead time (m)` | `min` |
| any step length that is not an exact whole number of seconds | `unknown` | `Lead time (u)` | - (tick labels are then just step indices) |
| `timedelta(milliseconds=500)` | `milliseconds` | `Lead time (m)` | `ms` |

`m` for minutes collides with the conventional metres/months, and `Lead time (u)` is meaningless. Every other call site (`neural_lam/models/module.py:780`, `:1019`) uses the full `time_step_unit` word; only this one truncates.

Display-only - no crash, no wrong numbers - but the `minutes` and `unknown` cases are both reachable with ordinary configs (a sub-hourly datastore, or any `step_length` that is not a whole number of seconds - `get_integer_time` returns `unknown` there).

## Reproduction

```python
from datetime import timedelta
import torch
from neural_lam import vis

# any BaseRegularGridDatastore whose step_length is timedelta(minutes=15)
fig = vis.plot_error_heatmap(torch.ones((4, 3)), datastore=datastore)
print(fig.axes[0].get_xlabel())   # "Lead time (m)"  -> want "Lead time (min)"
```

Not covered by existing tests - `tests/test_plotting.py` only exercises `timedelta(hours=1)`, for which `"hours"[0] == "h"` is coincidentally correct.

## Fix

Map the full unit name to a proper abbreviation (`minutes` -> `min`, `milliseconds` -> `ms`, `microseconds` -> `us`, ...), and label the `unknown` case `steps` since the tick values are then plain step indices.

**Comment by observingClouds** [comment]
Good catch! Thanks for opening a PR.
